src/System.Resources.Analyzers/Core/MarkAssembliesWithNeutralResourcesLanguage.cs

```diff
@@ -13,26 +16,124 @@ public abstract class MarkAssembliesWithNeutralResourcesLanguageAnalyzer : Diagn
+        protected const string StronglyTypedResourceBuilder = "StronglyTypedResourceBuilder";
+        private const string Designer = ".Designer.";
+
         private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(SystemResourcesAnalyzersResources.MarkAssembliesWithNeutralResourcesLanguageTitle), SystemResourcesAnalyzersResources.ResourceManager, typeof(SystemResourcesAnalyzersResources));
-        
+
         private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(SystemResourcesAnalyzersResources.MarkAssembliesWithNeutralResourcesLanguageMessage), SystemResourcesAnalyzersResources.ResourceManager, typeof(SystemResourcesAnalyzersResources));
         private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(SystemResourcesAnalyzersResources.MarkAssembliesWithNeutralResourcesLanguageDescription), SystemResourcesAnalyzersResources.ResourceManager, typeof(SystemResourcesAnalyzersResources));
-        
+
         internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Performance,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/bb385967.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
+        protected abstract void RegisterAttributeAnalyzer(CompilationStartAnalysisContext context, Action onResourceFound);
+
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+            analysisContext.EnableConcurrentExecution();
+
+            var hasResource = false;
+            analysisContext.RegisterCompilationStartAction(cc =>
+            {
+                RegisterAttributeAnalyzer(cc, () => hasResource = true);
```

Comment:
looked and symbol action doesnt give semantic model of the file where symbol is defined. (probably symbol can be defined in multiple files?)

so this should be cheaper.